16.0 upgrade doc condensing nama (#5953)
odoo/documentation: content/administration/upgrade.rst

:nosearch:
:show-toc:

=======
Upgrade
=======

Introduction to upgrades
------------------------

In Odoo, an upgrade is the process of moving your database from an older version of Odoo to a newer version. Each new version of Odoo comes with new features, bug fixes, security patches, and improvements **which is why uprading frequently is important**

Here are a few examples of supported upgrades:

* Odoo 13 to Odoo 16
* Odoo 14 to Odoo 16
* Odoo 15.3 to Odoo 16

Upgrading to an :doc:`unsupported version </administration/maintain/supported_versions>` is not possible.

.. note::
    An upgrade does not cover :

   * Downgrading to a previous version of Odoo (i.e., Odoo 15 to Odoo 12)
   * Changing :ref:`editions <install/editions>` (i.e., Community to Enterprise edition)
   * Switching :doc:`hosting type </administration/maintain/hosting_changes>` (i.e., On-Premise to Odoo Online or Odoo.sh)
   * Migration from another ERP to Odoo

.. important::
    If your database contain a custom module, :doc:`its source code must be upgraded </developer/reference/upgrade>` to the new version of Odoo before going through with the upgrade.

> In case of issues with the upgrade, you can contact the `support team <https://www.odoo.com/help>`_ for assistance.

Author: Sure but then the section would be like 
Spoiler block
note block
warning block
tip block, that seems too many blocks haha, but i'll apply the change

    :alt: Schema of the upgrade process : Export the database to a file, upload to the upgrade server, standard upgrade scripts, download the upgraded database, import the file into a database, (Optional : Custom upgrade scripts)
    :align: center

Where the standard upgrade scripts consists of a sequence of :ref:`reference/upgrade/migration-scripts`, each responsible for upgrading a specific part of a module's data.

.. note::
    You can always request a new upgraded test database in case you want to test the upgrade again or if you want to test the upgrade to a different version of Odoo.

.. tabs::

    .. group-tab:: Odoo Online

        .. _upgrade/request-test-database/odoo-online:

        Odoo databases can be manually upgraded directly from the main Odoo website. To upgrade an Odoo database, navigate to the `database manager page <https://www.odoo.com/my/databases>`_ page and sign in.

        The database manager page displays all of the Odoo databases associated with the user's account. Any databases that are not already on the most recent version of Odoo display an **arrow in a circle** icon next to the database name, indicating that the database can be upgraded.

>         .. image:: /administration/upgrade/odoo_online/databases-page.png

Author: I included your screenshot but I am scared that it does not display enough information, like where exactly is that screenshot supposed to show on the page, since there is no context surrounding it. It is still better than the previous one though so i'll include it

            :align: center
            :alt: The database manager page with an upgrade button next to the name of a database.

        Clicking on the **arrow in a circle** icon to display the following pop-up

        .. image:: /administration/upgrade/odoo_online/upgrade-pop-up.png
            :align: center
            :alt: The "Upgrade your database" pop-up.

        There are a few things to be filled on this pop-up :

>         - The version of Odoo you want to upgrade to (see :ref:`upgrade/which_version`)

Author: Yes, see screenshot sent above :D As stated in the (new) upgrade documentation, we can upgrade to any supported version and as you can see in https://www.odoo.com/documentation/master/administration/maintain/supported_versions.html the only supported version > 16.0 is saas 16.4, but if you were running Odoo 13 you could chose between 14, 15, 16 and 16.4 normally :D

    - Your server actions in the Action menu on form views as well as by selecting multiple records on list views

    Those are non-exhaustive lists that you can extend to your other Apps based on your use of Odoo.

In case of an issue with your test database, you can request the assistance of Odoo via the `support page <https://odoo.com/help>`_. They will be able to provide guidance on your upgrade or fix the issue. In any case, it is important to report any issue you encounter during your testing to Odoo so that they can be fixed before the upgrade of your production database.

Upgrading your production database
----------------------------------

Once you completed your :ref:`tests <upgrade/test_your_db>` and are confident that you can use your upgraded database as your main database without any issue, it is time to plan the Go-live day. During the upgrade of your production database, any modification done on it will not be saved. This is why we recommend not using your database during that time.

The Go-live day can be planned in coordination with the upgrade support analysts of Odoo reachable via the `support page <https://odoo.com/help>`_.

> .. important::

Author: What about merging the 2 paragraphs into this `Once the :ref:`tests <upgrade/test_your_db>` are completed, and you are confident that the upgraded database can be used as your main database without any issue, it is time to plan the go-live day. It can be planned in coordination with Odoo's upgrade support analysts, reachable via the `support page <https://odoo.com/help>`_.` ? (it will be in my commit so you'll be able to review at that time :D

    - employees being lost in the changes and new features
    - business interruptions (e.g., no longer having the possibility to validate an action)
    - poor customer experiences (e.g., an eCommerce website that does not work correctly)

The process of upgrading a database to production is similar to the one of upgrading a test database except for a few things

.. tabs::

    .. group-tab:: Odoo Online

        The process is equivalent to :ref:`upgrade/request-test-database` except for the purpose option which must be set to "Production" instead of "Test".

        .. important::
            Once the upgrade is requested, the database will be unavailable until the upgrade is finished. Once the process is completed, the database will be available in the new version of Odoo and there will be no way to revert to the previous version.

    .. group-tab:: Odoo SH

        The process is equivalent to :ref:`upgrade/request-test-database` except for that the "Production" branch must be selected before clicking the "Upgrade" tab

        .. image:: /administration/upgrade/odoo_sh/odoo-sh-prod.png
            :align: center
            :alt: View from the upgrade tab

>         The actual process is **triggered as soon as a new commit is added** to the branch. This allows the upgrade process to be synchronized with the change of the source code of the custom modules.

Author: Changed your change to `The actual process is **triggered as soon as a new commit is added** to the branch. This allows the upgrade process to be synchronized with the custom modules' modified source code.` to reflect more accurately the process, I hope that's okay:D

We always recommend upgrading **to the latest version of Odoo** to benefit from the latest features, improvements, and security fixes. If you are hosted on Odoo Online, you will automatically be enrolled in the rolling release program and will always be on the latest version of Odoo if your database can be upgraded automatically.

.. _upgrade_faq/rolling_release:

What is the Rolling Release (applicable to Odoo Online databases) ?
===================================================================

This feature allows customers to upgrade their database directly from a message prompt sent to the database administrator as soon as the new version is released. Odoo first tests the upgrade to the next version and if no issues are detected, the database administrator receives a message with 2 options:

#. To 'Upgrade Now', which immediately triggers the upgrade of your live production database. (Discouraged)

#. To take you to your `database manager <https://www.odoo.com/my/databases/>`_ where you can `request an upgraded test database <https://upgrade.odoo.com/#online/>`_ and check the upgraded test database for any discrepancies. (Recommended)

> If a database cannot be successfully upgraded automatically, then a ticket is automatically created to allow the upgrade technicians of Odoo to investigate the issue and fix it. Once the issue is fixed, the database is unblocked from the process and is available to be tested automatically again

Author: I heavily adapted this section with your comments, could you take another look :D ?

.. note::

>     In case of a database running a version that will soon be unsupported, the database can be automatically upgraded to the latest supported version of Odoo through the rolling release process.

Reviewer: ```suggestion
    If a database runs a version that will soon be unsupported, the database can be automatically upgraded to the latest supported version of Odoo through the rolling release process.
```
Do we have Odoo Online DBs running Odoo 14? If it's an unlikely/impossible scenario, we might want to remove that note.

Author: Yes, databases for which there is an issue with their upgrade (deactivated view, traceback during the upgrade process, ...)  I took this screenshot on a SaaS database currently running Odoo 14 and wishing to upgrade 
![image](https://github.com/odoo/documentation/assets/32837941/d954f983-a32b-4b35-85f6-404f86475d60)

> How is my data handled in the Upgrade Platform?

Reviewer: ```suggestion
Data protection during upgrades
```
This doesn't need to be a question

===============================================

> The Odoo Upgrade platform uses the same Privacy Policy as the rest of Odoo.com services.

Reviewer: Is that the upgrade.odoo.com page? Feels weird to call it a platform, to be honest.
Maybe link the privacy policy.

Author: It indeeds refer upgrade.odoo.com, as computer scientist I feel that it is not weird since a platform can refer to any (web) interface where you can submit data. Furthermore on the privacy policy page we can find a similar reference https://www.odoo.com/privacy#:~:text=legacy%20web%2Dbased%20upgrade%20platform%2C :D

Your data is hosted on servers that follow our security guidelines, namely:

- SSL - All web connections to client instances are protected with 256-bit SSL encryption (HTTPS with a 2048-bit modulus SSL certificate), and running behind Grade A SSL stacks. All our certificate chains are using SHA-2 already.
- Safe System - Our servers are running recent Linux distribution with up-to-date security patches, with firewall and intrusion countermeasures (not disclosed for obvious reasons).

Servers are located at the same locations as our Cloud providers with the following services:

- Restricted perimeter, physically accessed by authorized data center employees only
- Physical access control with security badges or biometrical security
- Security cameras monitoring the data center locations 24/7
- Security personnel on-site 24/7

The uploaded and migrated databases uploaded to the Upgrade platform are kept for up to 3 months and are permanently deleted following that period.

You can learn more about privacy and data handling at Odoo by visiting our `General Data Protection Regulation page <https://www.odoo.com/gdpr>`_.

> Why does some issues require a developer from Odoo ?

Reviewer: Merge with the info on custom modules.

====================================================

The upgrade process is split in 2 parts : the standard upgrade process, and the upgrade of the custom modules. If some important data is removed during the standard upgrade process, a migration script must be injected to save them. Since the standard migration scripts are executed on a server owned by Odoo S.A., only trusted code can be injected in the process.

Custom migration scripts are of no use in those situations since the data is removed before the custom migration scripts are executed. In situation where the standard upgrade process does not complete due to an error, the custom migration scripts are not executed either.

> Why can't you make my view be exactly like it was before ?

Reviewer: Merge with the info on testing.

      - The **version** of Odoo you want to upgrade to, usually the latest version
      - The **email** address that should receive the link to the upgraded database
      - The :guilabel:`Purpose` of the upgrade, which is automatically set to :guilabel:`Test` for
        your first upgrade request

      .. image:: upgrade/upgrade-popup.png
         :alt: The "Upgrade your database" popup.

      The :guilabel:`Upgrade in progress` tag is displayed next to the database name until
      completion. Once the process succeeds, an email containing a link to the upgraded test
      database is sent to the address provided. The database can also be accessed from the database
      manager by clicking the dropdown arrow before the database name.

      .. image:: upgrade/access-upgraded-db.png
         :alt: Clicking the menu arrow displays the upgraded test database.

   .. group-tab:: Odoo.sh

      Odoo.sh is integrated with the upgrade platform to simplify the upgrade process.

      .. note::
         The :guilabel:`Upgrade` tab is available for valid projects with a valid production build

>          in the branches view.

Author: Part of the current documentation : https://www.odoo.com/documentation/16.0/administration/upgrade/odoo_sh.html#overview

I think it refers to the branch being up and running

>          in the branches view.

Reviewer 2: IMO, it was meant for:
"valid projects": projects with a valid subscription (that are still paying), so for example no trial subscriptions, or educational ones (don't know if they exist on SH)
and "valid production build": what you said.

But it's still seems confusing.

I think we could ask STS (he developed the feature and knows the conditions he added in the code)

Hi @sts-odoo 
Could you help us understand what you meant when you wrote this note in the documentation?
![image](https://github.com/odoo/documentation/assets/28193899/8272b8f5-5757-4a36-bfd0-2eea1213f39e)
What does "valid project" and "valid production build" means?

>          in the branches view.

Reviewer 3: @augustoPerez @nmarotte I confirm that it means a valid project (running  odoo.sh subsrciption on odoo.com not a trial one, ie partnership), AND a valid subcription for the database (running based on odoo.com status).

In other places in the UI of Odoo.sh customer will see if their database is valid or not and why so it is clear enough. The documentation should be complementary to the UI not a copy or replacement of the messages that exist already in the UI.

@augustoPerez does the upgrade platform accepts non valid database to upgrade? If so, why, and do we want that as well on odoo.sh?

>          in the branches view.

Author: Hey @sts-odoo , indeed the upgrade platform only accepts database with a valid subscription number, so no subscription, no request. I changed the note to ```The :guilabel:`Upgrade` tab is available for projects with a valid subscription.´´´ since, indeed, if the production branch is red/killed/dropped, it would make no sense to upgrade it. Thanks for your input :D

      .. image:: upgrade/odoo-sh-staging.png
         :alt: Odoo.sh project and tabs

      The **latest production daily automatic backup** is then sent to the `upgrade platform
      <https://upgrade.odoo.com>`_.

      Once the upgrade platform is done upgrading the backup and uploading it on the branch, it is
      put in a **special mode**: each time a **commit is pushed** on the branch, a **restore
      operation** of the upgraded backup and an **update of all the custom modules** occur. This
      allows you to iterate on your custom modules' upgrade scripts quickly. The log file of the

>       upgrade process can be found in your newly upgraded staging build by going to

Author: Sure, this was part of the "bullet-proof" documentation that is currently written here https://www.odoo.com/documentation/16.0/administration/upgrade/odoo_sh.html

      :file:`~/logs/upgrade.log`.

      .. note::
         Most Odoo.sh users have custom modules installed on their database. In that case, the
         modules' source code must be up-to-date with the target version of Odoo before the

>          upgrade can be performed. Check out the :doc:`upgrade for developers' documentation

Author: I took your suggestion into account and changed the phrasing. It is not recommended to use the 2nd form ("you") when writing, so I changed that. I also added a bit specifying that this applies if there are custom modules

         </developer/reference/upgrade>` for more information. In addition, if a module is not
         needed after an upgrade, :ref:`you can remove customizations

>          <upgrade/remove_customizations>`.

Author: Instead of having 2 notes one after the other, I merged it with the previous one, what do you think ?

      .. code-block:: console

         $ python <(curl -s https://upgrade.odoo.com/upgrade) --help

      An upgraded test database can also be requested via the `Upgrade page
      <https://upgrade.odoo.com/>`_.

      .. note::
         - For security reasons, only the person who submitted the upgrade request can download it.
         - For storage reasons, the database's copy is submitted without a filestore to the upgrade
           server. Therefore, the upgraded database does not contain the production filestore.
         - Before restoring the upgraded database, its filestore must be merged with the production
           filestore to be able to perform tests in the same conditions as it would be in the new
           version.
         - The upgraded database contains:

           - A `dump.sql` file containing the upgraded database
           - A `filestore` folder containing files extracted from in-database records into
             attachments (if there are any) and new standard Odoo files from the targeted Odoo
             version (e.g., new images, icons, payment provider's logos, etc.). This folder be

>              merged with the production filestore to get the full upgraded filestore.

Author: Indeed I don't know why there is a trailing ``` in your commit so I will add it via my commit instead of committing your suggestion directly

.. note::
   You can request multiple test databases if you wish to test an upgrade more than once.

In all cases, the process is the same:

#. Exporting the database to a file
#. Uploading the file to the upgrade server
#. Running a series of :ref:`standard migration scripts <reference/upgrade/migration-scripts>` to
   upgrade standard modules
#. Downloading the upgraded database
#. Importing the file into a database
#. (*Optional*) Running a series of :ref:`custom migration scripts

>    <reference/upgrade/migration-scripts>` developed by third parties to upgrade custom modules

Reviewer 2: I don't think it's clear when, why or what runs

Author: It is a numbered list, so the when is : after the previous step
I am not sure that "why" is important, but I can add "to upgrade them" (see full sentence below)
As for the what, yes I agree

full sentence : ```#. (*If applicable*) :ref:`Custom migration scripts <reference/upgrade/migration-scripts>`   developed by the maintainer of your custom modules are ran to upgrade them.```

What do you think ?

>    <reference/upgrade/migration-scripts>` developed by third parties to upgrade custom modules

Reviewer 2: It could be nice to give an overview of what we're gonna talk about, like a table of contents, how the process work or somthing like that, not sure how yet. But as it is right now, I'm not sure what's the added value of this here.

Author: IMO it helps to understand what is going on exactly behind the scenes when an upgrade happen, since it is different than the usual "update" of many applications (Word, web browser, Android/iOS app) where the new code is downloaded and the application restart. With Odoo, the data is transmitted to an other computer, it is changed significantly, and then uploaded back to the sender.

This can help external to have a better understanding on the migration (movement) of their (sensible) data

But you are not the only one raising doubts about this section, if you really think that having this numbered list will hurt us more than it will do good, I will remove it, but IMO it does not have a negative impact:  might be a positive impact, or a neutral one

Reviewer 2: I'm thinking that a mix of your section and this one in the current doc: https://www.odoo.com/documentation/16.0/administration/upgrade.html#process-workflow
could be a bit more helpful
Talking about the whole upgrade process beginning to end and what happens in the middle with the upgrade platform. However, we should find a way to have it clear and not repeat ourselves (as it describes both testing and prod upgrades, and the "platform" process is the same for both), and also check where to add it so it's clear and not confusing :thinking: 

I suggest to leave this for after the OXP, to have more time to find the best place and words for it.

Could you remove this for the moment?
Of course, if you have some ideas already you can comment them or make a skectch for the future (not a priority before publishing the rest of the doc)

Author: I think we are barking at the wrong tree, this is useful but not on administration/upgrade -> I am moving it to the developer page for later use :D

Reviewer 2: Sounds good to me !

.. _upgrade/upgrade_report:

.. note::

>    When an upgrade request is completed, an upgrade report is emailed and made available in the
>    Discuss app. It contains important information about the changes introduced by the new version.

Reviewer 2: ```suggestion
   When an upgrade request is completed, an upgrade report is appended to the successful upgrade
   email and it becomes available in the Discuss app for users who are part of the "Administration
   / Settings" group. This report provides important information about the changes introduced by the
   new version.
```

Author: I agree but "attached" and not "appended" would sound better IMO, adding it as my own commit to do the change

.. _upgrade/test_your_db:

Testing the new version of the database
---------------------------------------

> It is essential to spend some testing the upgraded test database to ensure that you are not stuck in

Author: Nice catch, got it

your day-to-day activities by a change in views, behavior, or an error message once the upgrade goes
live.

.. note::
   Test databases are neutralized, and some features are disabled to prevent them from impacting the
   production database:

   #. Scheduled actions are disabled.
   #. Outgoing mail servers are disabled by archiving the existing ones and adding a fake one.
   #. Payment providers and delivery carriers are reset to the test environment.

Testing as many of your business flows as possible is strongly recommended to ensure they are
working correctly and to get more familiar with the new version.

> .. admonition:: Basic test checklist

Author: Good idea

   - Checking if the route to receive this product is the same in your production database
     (Inventory app).
   - Selling this product (Sales app) to a random customer.
   - Opening your customer database (Contacts app), selecting a customer (or company), and checking
     its data.
   - Shipping this product (Inventory app).
   - Checking if the route to ship this product is the same as in your production database
     (Inventory app).
   - Validating a customer invoice (Invoicing or Accounting app).
   - Crediting the invoice (issuing a credit note) and checking if it behaves as in your production
     database.
   - Checking your reports' results (Accounting app).
   - Randomly checking your taxes, currencies, bank accounts, and fiscal year (Accounting app).
   - Making an online order (Website apps) from the product selection in your shop until the
     checkout process and checking if everything behaves as in your production database.

   This list is **not** exhaustive. Extend the example to your other apps based on your use of Odoo.

If you face an issue while testing your upgraded test database, you can request the assistance of

> Odoo via the `support page <https://odoo.com/help>`_. In any case, it is essential to report any

Author: Personally, I think that if the customer clicks on this link after reading that paragraph, because he encounters an issue with the testing of his upgraded test database, and that he read `an issue while testing your upgraded test database`, he would naturally check the option (currently named as) ` An issue related to my future upgrade (I am testing an upgrade)`

Having a screenshot would add yet another image to maintain throughout the year when name changes, as you can see even in the current documentation, the screenshot does not match exactly the current selection field of odoo.com/help.

IMO, I think there will be little to no user that will read that sentence, click the button, and select another option

testing. Those changes cannot be reverted on a case-by-case basis. However, if a change introduced
by a new version breaks a customization, it is the responsibility of the maintainer of your custom
module to make it compatible with the new version of Odoo.

.. tip::
   Do not forget to test:

   - Integrations with external software (EDI, APIs, etc.)
   - Workflows between different apps (online sales with eCommerce, converting a lead all the way to
     a sales order, delivery of products, etc.)
   - Data exports
   - Automated actions
   - Server actions in the action menu on form views, as well as by selecting multiple records on
     list views

Upgrading the production database
---------------------------------

Once the :ref:`tests <upgrade/test_your_db>` are completed and you are confident that the upgraded
database can be used as your main database without any issue, it is time to plan the go-live day. It
can be planned in coordination with Odoo's upgrade support analysts, reachable via the `support page
<https://odoo.com/help>`_.

.. important::
   - Any modification to your production database will **not** be saved during the upgrade of your

>      production database. This is why we recommend not using it during that time.

Author: You're right. I think we could even remove the whole sentence about modifications not saved ? Since even for On-Premise, it is logical that once the dump has been uploaded, any modification done after the upload will not be taken into account.

What do you think ? Should I simply move it to On-Premise or remove it entirely ?

>      production database. This is why we recommend not using it during that time.

Reviewer 2: It is logical for us, but maybe not for all the customers
I would move it into the On-Premise tab just in case some customer's are not aware

   - Going into production without first testing may lead to:

     - Users failing to adjust to the changes and new features
     - Business interruptions (e.g., no longer having the possibility to validate an action)
     - Poor customer experience (e.g., an eCommerce website that does not work correctly)

The process of upgrading a production database is similar to upgrading a test database with a few
exceptions.

.. tabs::

   .. group-tab:: Odoo Online

      The process is equivalent to :ref:`obtaining an upgraded test database
      <upgrade/request-test-database>`, except for the purpose option, which must be set to
      :guilabel:`Production` instead of :guilabel:`Test`.

      .. warning::
         Once the upgrade is requested, the database will be unavailable until the upgrade is
         finished. Once the process is completed, it is impossible to revert to the previous
         version.

   .. group-tab:: Odoo.sh

      The process is equivalent to :ref:`obtaining an upgraded test database
      <upgrade/request-test-database>`, except that the :guilabel:`Production` branch must be

>       selected before clicking the :guilabel:`Upgrade` tab.

Author: `except that the :guilabel:`Production` branch must be selected before clicking the :guilabel:`Upgrade` tab.`
-> 
`on the :guilabel:`Production` branch` 

What do you think ?

>       selected before clicking the :guilabel:`Upgrade` tab.

Reviewer 2: Maybe I'd say: `The process is similar to...` instead of `equivalent`
For me (and I'd rather avoid confusion for other people reading the doc as well), "equivalent" means that it's doing the same.
While in fact, even if the button is the same, it's not triggered by the same factors (as it's explained after).
That's why I would prefer "similar", so it underlines that it's not the same, it might resemble, but it's not the same.

Also in the change you put ```on the :guilabel:`Upgrade` branch.``` :see_no_evil:

      .. image:: upgrade/odoo-sh-prod.png
         :alt: View from the upgrade tab

      The actual process is **triggered as soon as a new commit is added** to the branch. This
      allows the upgrade process to be synchronized with the deployment of the custom modules'

>       upgraded source code.

Author: This could be summarized as "If there are no custom modules, the upgrade process is triggered immediately." at the end of the paragraph, what do you think ?

      .. important::
         The database is unavailable throughout the process. If anything goes wrong, the platform
         automatically reverts the upgrade, as it would be for a regular update. In case of success,
         a backup of the database before the upgrade is created.

      The update of your custom modules must be successful to complete the entire upgrade process.
      Make sure the status of your staging upgrade is :guilabel:`successful` before trying it in
      production. More information on how to upgrade your custom modules can be found in the
      :ref:`upgrading customizations documentation <upgrade/upgrading_customizations>`.

   .. group-tab:: On-premise

      The command to upgrade a database to production is similar to the one of upgrading a test
      database except for the argument `test`, which must be replaced by `production`:

      .. code-block:: console

>          $ python <(curl -s https://upgrade.odoo.com/upgrade) production -d <your db name> -t <target version>

Author: There is no difference in the source code, I think it is due to 'test' being some kind of reserved/key word. I'm not sure I can do anything about it

      An upgraded production database can also be requested via the `Upgrade page
      <https://upgrade.odoo.com/>`_.

      .. important::
         When requesting an upgraded database for production purposes, the copy is submitted without
         a filestore. Therefore, the upgraded database filestore must be merged with the production
         filestore before deploying the new version.

In case of an issue with your production database, you can request the assistance of Odoo via the

> `support page <https://odoo.com/help>`_.

Author: Same for the other one, when reading this paragraph, the only option that make sense for the user is to select "An issue related to my upgrade (production)", I think it would be redundant to have the image, as it is recommended that the documentation be kind of "minimalist"

However, I can pre-select the correct option with the URL, in this case with `https://odoo.com/help?stage=post_upgrade`, and `https://odoo.com/help?stage=migration` (feel free to try those URL to confirm it works haha)

What do you think ?

> `support page <https://odoo.com/help>`_.

Reviewer 2: I like the idea of putting the link with the preselected option, for both test and post upgrade issues :+1: 

N.B.: Even if something makes sense and seems logic, don't asume customers are gonna realise or follow them haha

  - :doc:`Changing editions <maintain/enterprise>` (e.g., from Community to Enterprise)
  - :doc:`Switching hosting type </administration/maintain/hosting_changes>` (e.g., from on-premise
    to Odoo Online).
  - Migrating from another ERP to Odoo

.. warning::
   If your database contains a **custom module**, you must :doc:`upgrade its source code first
   </developer/reference/upgrade>` to the new version of Odoo **before upgrading**.

Upgrading in a nutshell
-----------------------

#. Request an upgraded test database (see :ref:`obtaining an upgraded test database
   <upgrade/request-test-database>`).

#. Thoroughly test the upgraded database (see :ref:`testing the new version of the database
   <upgrade/test_your_db>`).

#. Report any issue encountered during the testing to Odoo via the `support page
   <https://odoo.com/help?stage=migration>`__.

#. (If applicable) : upgrade the source code of your custom module to be compatible with the new
   version of Odoo.

#. Once all issues are resolved and you are confident that the upgraded database can be used as
   your main database without any issue, plan the upgrade of your production database.

> #. You or Odoo upgrade the production database, rendering it unavailable for a few hours
>    (see :ref:`upgrading the production database <upgrade/upgrade-prod>`).

Reviewer 2: I'd say it similar to how it is in the test part
Also, let's not say it takes hours :see_no_evil: 
I know it's already there in the current doc, but it can scare people and it's usually not true, most dbs are small and take at most 15-20 mins (tho I wouldn't say it either as each db is different).
```suggestion
# Request the upgraded for the production database, rendering it unavailable for the time it takes to complete the process (see :ref:`upgrading the production database <upgrade/upgrade-prod>`).
```
For the rest, this new section looks good to me :+1:
Thanks for adding it back and adapt it to the new doc schema!!
